app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart

```diff
@@ -198,10 +198,12 @@ class CheckForWaitingPullRequests extends ApiRequestHandler<Body> {
       final String sha = commit['oid'] as String;
       final List<Map<String, dynamic>> statuses =
           (commit['status']['contexts'] as List<dynamic>).cast<Map<String, dynamic>>();
-      final List<Map<String, dynamic>> checkRuns =
-          (commit['checkSuites']['nodes'].single['checkRuns']['nodes'] as List<dynamic>).cast<Map<String, dynamic>>();
+      List<Map<String, dynamic>> checkRuns;
+      if (commit['checkSuites']['nodes'] != null && (commit['checkSuites']['nodes'] as List<dynamic>).isNotEmpty) {
+        checkRuns =
```

Comment:
Nit: I think the if condition could also use `?.` to simplify.

Reply:
Unfortunately ?. didn't work for too many nested maps.